src/transformStateWithClones.js

```diff
@@ -7,7 +7,35 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arrToReturn = [];
+  // copying state
+  let stateCopy = { ...state };
+
+  // iterate throught actions array
+  for (const action of actions) {
+    const { type, extraData, keysToRemove } = action;
+
+    if (type === 'addProperties') {
+      // Add to already existing state extraData.
+      stateCopy = { ...stateCopy, ...extraData };
+    }
+
+    if (type === 'removeProperties') {
+      // This one is easy to understand actually
+      for (const key of keysToRemove) {
+        delete stateCopy[key];
+      }
+    }
+
+    // Reset stateCopy, like IPhone
+    if (type === 'clear') {
+      stateCopy = {};
+    }
+
+    arrToReturn.push({ ...stateCopy });
```

Comment:
[DONT REPEAT YOURSELF] - Since you are pushing a copy of `stateCopy` to `arrToReturn` at the end of each loop iteration, consider moving this operation outside of the `switch` block but still inside the loop, after the current action is processed. This will adhere to the DRY principle and avoid repeating the spread operation for each case.